Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/PrevUntilOperation.php

```diff
@@ -70,22 +71,21 @@ public function evaluate(FlowQuery $flowQuery, array $arguments): void
     {
         $output = [];
         $outputNodeIdentifiers = [];
-        $until = [];
 
         foreach ($flowQuery->getContext() as $contextNode) {
-            $prevNodes = $this->getPrevForNode($contextNode);
+            $prevNodes = $this->contentRepositoryRegistry->subgraphForNode($contextNode)
+                ->findPrecedingSiblingNodes(
+                    $contextNode->nodeAggregateId,
+                    FindPrecedingSiblingNodesFilter::create()
+                );
             if (isset($arguments[0]) && !empty($arguments[0])) {
                 $untilQuery = new FlowQuery($prevNodes);
                 $untilQuery->pushOperation('filter', [$arguments[0]]);
-
-                $until = $untilQuery->getContext();
+                $untilNodes = Nodes::fromArray(iterator_to_array($untilQuery));
             }
-            /** @var array<int,mixed> $until */
-
-            if (isset($until[0]) && !empty($until[0])) {
-                $prevNodes = $prevNodes->until($until[0]);
+            if (isset($untilNodes) && !$untilNodes->isEmpty()) {
+                $prevNodes = $prevNodes->previousAll($untilNodes->first())->reverse();
```

Comment:
i dont fully grasp this line, and in `NextAllUntil` we use

```
$nextNodes = $nextNodes->previousAll($untilNodes->first());
```

but it seems to work :D

Reply:
Was needed to get the order we had before. Can check the priorities once more.

Reply:
I suggest to discuss about changing the order later. Since this does nothing bad it could be merged for beta 1.